src/DataDictionary/graph/GraphCalculator/graphCalculatorHelper.js

```diff
@@ -0,0 +1,520 @@
+import Viz from 'viz.js';
+import { Module, render } from 'viz.js/full.render';
+import _ from 'underscore';
+import { createNodesAndEdges, createDotStrinByNodesEdges } from '../../../GraphUtils/utils';
+import { truncateLines, graphStyleConfig } from '../../utils';
+import { getCategoryColor } from '../../NodeCategories/helper';
+
+export const getAllTypes = (nodes) => {
+  const types = nodes.map(node => node.type);
+  const uniqueTypes = _.uniq(types);
+  return uniqueTypes;
+};
+
+export const getAllChildrenNodeIDs = (highlightingNode, nodes) => {
```

Comment:
In general, could you put some comments in this file so that we can understand more about the logic that you use to compute the position of nodes and edges.
You can also name the condition and some specific calculating functions meaningful name to increase the readability.